packages/eui/.storybook/addons/code-snippet/decorators/utils.ts

```diff
@@ -0,0 +1,218 @@
+https://github.com/storybookjs/storybook/blob/2bff7a1c156bbd42ab381f84b8a55a07694e7e53/code/renderers/react/src/docs/jsxDecorator.tsx#L224 */
+
+import { ReactElement, FunctionComponent, ComponentType } from 'react';
+import * as prettier from 'prettier';
+import tsParser from 'prettier/parser-typescript';
+import { StoryContext } from '@storybook/react';
+
+// @ts-ignore - config import
+import basePrettierConfig from '../../../../.prettierrc';
+
+export const toPascalCase = (str: string) =>
+  str.charAt(0).toUpperCase() + str.slice(1);
+
+/* Helpers for React specific checks */
+export const isMemo = (component: any) =>
+  component.$$typeof === Symbol.for('react.memo');
+export const isForwardRef = (component: any) =>
+  component.$$typeof === Symbol.for('react.forward_ref');
+export const isFragment = (component: any) =>
+  component.type?.toString().includes('fragment') ||
+  component.$$typeof?.toString().includes('fragment');
+
+/* Helpers */
+// returns the displayName and handles typing as
+// otherwise `type` would not be typed
+export const getElementDisplayName = (
+  node: ReactElement<any>
+): string | undefined => {
+  let displayName;
+
+  if (typeof node.type === 'function' || typeof node.type === 'object') {
+    displayName =
+      (node.type as FunctionComponent).displayName ??
+      (node.type as FunctionComponent).name ??
+      undefined;
+  }
+
+  return displayName;
+};
+
+// returns the displayName after resolving Emotion wrappers
+export const getEmotionComponentDisplayName = (
+  node: ReactElement<any>
+): string | undefined => {
+  const displayName = getElementDisplayName(node);
+
+  if (
+    (typeof displayName === 'string' && displayName.startsWith('Emotion')) ||
+    node.props?.__EMOTION_TYPE_PLEASE_DO_NOT_USE__ != null
+  ) {
+    const { __EMOTION_TYPE_PLEASE_DO_NOT_USE__: emotionData } = node.props;
+    const isForwardRefComponent = isForwardRef(emotionData);
+
+    // we need to rely here on the reference Emotion stores to know what component this actually is
+    const replacementName = isForwardRefComponent
+      ? emotionData.__docgenInfo.displayName
+      : typeof emotionData === 'string'
+      ? emotionData
+      : emotionData?.displayName;
+
+    // remove internal component underscore markings
+    return replacementName ? replacementName.replace('_', '') : displayName;
+  }
+
+  return displayName;
+};
+
+export const getComponentDisplayName = (
+  context: StoryContext | undefined
+): string | undefined => {
+  if (!context) return undefined;
+  const component = context.component as ComponentType<any> & {
+    __docgenInfo?: { displayName?: string };
+  };
+
+  return component?.displayName ?? component.__docgenInfo?.displayName;
+};
+
+export const isEmotionComponent = (node: ReactElement<any>): boolean => {
+  const displayName = getElementDisplayName(node);
+  const matches =
+    typeof displayName === 'string'
+      ? displayName.match(/^(Emotion)(\w)*/g)
```

Comment:
It looks like this one is still here?
```suggestion
      ? displayName.startsWith('Emotion')
```

Reply:
Huh, weird cause I was sure I changed it. I'll update again! 😅

Reply:
Updated in this [commit](https://github.com/elastic/eui/pull/7716/commits/118adc8eed0f4a85d7eac509d9c32fa65d666e2d)